core/src/repair_generic_traversal.rs

```diff
@@ -125,9 +125,21 @@ pub fn get_closest_completion(
                 let shred_index = blockstore.get_index(slot).unwrap();
                 let dist = if let Some(shred_index) = shred_index {
                     let shred_count = shred_index.data().num_shreds() as u64;
-                    last_index - shred_count
+                    debug_assert!(
+                        last_index + 1 >= shred_count,
+                        "last_index={} shred_count={}",
+                        last_index,
+                        shred_count
+                    );
+                    (last_index + 1).saturating_sub(shred_count)
```

Comment:
Can we make the addition saturating while we're here?

Reply:
done